src/main/java/seedu/address/ui/PersonCard.java

```diff
@@ -75,5 +80,37 @@ public PersonCard(Person person, int displayedIndex) {
                 .forEach(tag -> tags.getChildren().add(new Label(tag.tagName)));
         uniqueId.setText(person.getId().map(x -> Integer.toString(x)).orElse("NONE"));
 
+        int numberOfNotes = person.getNotes().size();
+        notesButton.setText("Notes (" + numberOfNotes + ")");
+
+    }
+
+    /**
+     * Opens a new window to display the notes of the person.
+     */
+    @FXML
+    public void handleNotesButtonClick() {
```

Comment:
Basically, I'm unable to get JavaFX to create a new window displaying the notes (using the classes that have been defined in the project causes errors when I construct them). I've hacked around it by using an Alert, but if we want any more logic beyond just displaying text, we'll have to fix this at some point.

Reply:
I've traced the issue, it's an issue with your FXML structure.
Let's fix this and use this approach instead of the hacky alert box method.

```diff
--- a/src/main/resources/view/NotesWindow.fxml
+++ b/src/main/resources/view/NotesWindow.fxml
@@ -7,16 +7,25 @@
 <?import javafx.scene.control.Button?>
 <?import javafx.scene.control.Label?>
 <?import javafx.geometry.Insets?>
-<VBox xmlns="http://javafx.com/javafx/17" xmlns:fx="http://javafx.com/fxml/1" spacing="10" alignment="CENTER">
-    <padding>
-        <Insets top="10" right="10" bottom="10" left="10" />
-    </padding>

-    <Label text="Notes" style="-fx-font-weight: bold; -fx-font-size: 18px;" />
+<?import javafx.scene.Scene?>
+<fx:root resizable="false" title="Notes"
+         type="javafx.stage.Stage" xmlns="http://javafx.com/javafx/17" xmlns:fx="http://javafx.com/fxml/1">
+    <scene>
+        <Scene>
+            <VBox spacing="10" alignment="CENTER">
+                <padding>
+                    <Insets top="10" right="10" bottom="10" left="10" />
+                </padding>

-    <ScrollPane fitToWidth="true" prefHeight="300" VBox.vgrow="ALWAYS">
-        <ListView fx:id="notesListView" />
-    </ScrollPane>
+                <Label text="Notes" style="-fx-font-weight: bold; -fx-font-size: 18px;" />

-    <Button text="Close" />
-</VBox>
+                <ScrollPane fitToWidth="true" prefHeight="300" VBox.vgrow="ALWAYS">
+                    <ListView fx:id="notesListView" />
+                </ScrollPane>
+
+                <Button text="Close" />
+            </VBox>
+        </Scene>
+    </scene>
+</fx:root>
```